### DISCUSSION TITLE
Npx tailwindcss  results in "Unexpected Token ."  This error is unhelpful.

### DISCUSSION THREAD

**Discussion by severian1778** [tailwindlabs/tailwindcss#8807] [Help]
As described in title.

This is on up to date npm and tailwind build.

Its kind of hard to diagnose.

**Comment by adamwathan** [comment]
Hey! Gonna need a lot more information here sorry. Can you provide exact steps to reproduce as well as provide the complete error?

Just a few examples of things I don't know from this:

- Did you install Tailwind before running npx, or just trying to run npx without installing Tailwind locally?
- What operating system are you using?
- What version of node?
- What command specifically did you actually run? `npx tailwindcss init`? `npx tailwindcss -o output.css`?

So please provide every detail, we have issue templates for a reason 🙂

**Reply by Bolaji5821** [reply]
npx postcss

**Reply by Inori-dokusha** [reply]
- install tailwind after run `npx init`
- i'm install tailwind in android with termux
- node version 21.6.2
- `npx tailwindcss -i ./input.css -o ./output.css --watch`

**Comment by Keyskeeper** [comment]
Hello,
I have a similar issue on **Ubuntu 20.04**

- Node v12.22.12
- npm 6.14.16

Running `npx tailwindcss -o output.css` fails with a message :
`Unexpected token '.'`

I tried to update or reinstall  nodejs and tailwindcss but I still have the same issue

**Reply by Thosine-01** [reply]
I'm also having similar error on ubuntu 20.04

npm version 8.5.1

Running **npx tailwindcss init** gives this error :+1: "**/home/lawal/Desktop/Tailwind/lesson1/node_modules/yaml/dist/compose/composer.js:33
                if (prelude[i + 1]?.[0] !== '#')"**

**Reply by Arsalanamin404** [reply]
Same error if (prelude[i + 1]?.[0] !== '#')"

**Reply by Asikul001** [reply]
Same problem comes in ubuntu 20.04.
plz anyone solve this problem.

**Comment by Charlygnusa** [comment]
Hello,
Me too have a similar issue on Ubuntu 18.04

Node v14.15.0
npm 6.14.8
Running "npx tailwindcss init -p" in new project fails with a message :
Unexpected token '.'

**Comment by COValhalla** [comment]
Same issue here. Following either the TailwindCSS CLI or PostCSS instructions on the website. I've reverted back to 3.1.4 and everything is working as expected.

Running npx tailwindcss init

/home/mojo/TOP/weather-app/node_modules/yaml/dist/compose/composer.js:33
                if (prelude[i + 1]?.[0] !== '#')
                                   ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/mojo/TOP/weather-app/node_modules/yaml/dist/index.js:3:16)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)

**Reply by Keyskeeper** [reply]
works for me too with 3.1.4 fall back.

**Comment by Shaco74** [comment]
I had the exact same error message running _node 16.18.0, on Windows 11_.
I managed to fix the issue by **downgrading to node 16.13.2** using [nvm](https://github.com/nvm-sh/nvm). 

```sh
PS C:\Users\Kacpe\Documents\GitHub\SWE\SWE-MehrMarkt\Frontend> nvm use 16.13.2
Now using node v16.13.2 (64-bit)
PS C:\Users\Kacpe\Documents\GitHub\SWE\SWE-MehrMarkt\Frontend> npx tailwindcss init -p

Created Tailwind CSS config file: tailwind.config.js
Created PostCSS config file: postcss.config.js
```

**Comment by r01al** [comment]
I had this issue too.
now the issue is when using `npx tailwindcss`. `npx` will take the latest tailwind which working on higher node version.
so you don't need to upgrade/downgrade node version just use the proper tailwind version.
e.g `npx tailwindcss@3.1.8`

and i will work magicly!

**Reply by khaledYS** [reply]
brooooo, you saved my life. I was running on a problem and thought I would quit programing for a while.

**Reply by tskxz** [reply]
i tried with that
i got the same error
```
npx tailwindcss@3.1.8 init -p
/home/tsk/Desktop/eco-escola/node_modules/yaml/dist/compose/composer.js:33
                if (prelude[i + 1]?.[0] !== '#')
                                   ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/tsk/Desktop/eco-escola/node_modules/yaml/dist/index.js:3:16)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)

```

**Reply by prajwalppramod** [reply]
this worked. thanks

**Reply by juniormiranda89** [reply]
this worked. thank you very much

**Reply by Arivald8** [reply]
This worked for me too on Ubuntu 22.04.2 LTS, npm/npx@8.5.1. 

Thank you.

**Reply by The-Lone-Druid** [reply]
This one saved the day.

**Reply by Putriamandahaifa** [reply]
You save my day, Thanks alot!

**Comment by Thosine-01** [comment]
Good pm tanjil. To debug this code all you need to do is to install the
latest node version.

**Reply by tskxz** [reply]
It worked, thanks

**Comment by codemewel** [comment]
Hello, I got the same error on my Ubuntu 22.04 when running npx tailwondcss init. I appreciate any help

**Reply by codemewel** [reply]
I finally fixed my issue. I just did purge the node and npm via terminal and install the node in snap store! It works perfectly fine now!

**Reply by Asikul001** [reply]
"unexpected token ."- this problem comes in ubuntu with tailwind version 3.4.1 so downgrade the tailwind version to 3.1.8 and then run npx tailwindcss init. i think this will work in ubuntu

**Reply by RanitManik** [reply]
Yes, this one worked for me too (Ubuntu 22.04 LTS).

**Reply by ilyaskhan15** [reply]
**It is conflict in version...try to install the lastest version of nvm**

**Comment by Jetsadakorn-s** [comment]
Delete the **node_modules** directory and the package-lock.json file (or yarn.lock if you're using Yarn) from your project's root directory. Then, reinstall the dependencies by running **npm install** or **yarn install**. See if the version of both node and tailwind are supported.

**Comment by RamkumarMrj** [comment]
When attempting to initialize Tailwindcss, I encountered an issue.

```
PS C:\> npx tailwindcss -o output.css
Unexpected token .
(node:22988) ExperimentalWarning: The fs.promises API is experimental

PS C:\> npx tailwindcss init
Unexpected token .
(node:25848) ExperimentalWarning: The fs.promises API is experimental
```


I'm currently using a specific project version. I would like to know if this version supports the latest Tailwindcss or not.


```
PS C:\> node -v
v10.13.0

PS C:\> npm -v
6.4.1

PS C:\> ng v 
Angular CLI: 11.1.4
```

**Comment by Gelloiss** [comment]
Hey!
Use actual node version, and all will be cool :)
you forgot to change the node version to 18))

**Comment by JoAzar** [comment]
El problema que todos tienen es que no instalaron node como señaló Gelloiss, acá hay una indicación de cómo hacerlo en freeCode: 

Una vez instalado node deben actualizarlo con nvm install 14 (14 o la versión que quieran)

The problem everyone has is that they didn't install node as Gelloiss pointed out, here is an indication of how to do it in freeCode:

Once node is installed, you must update it with nvm install 14 (14 or the version you want)

[https://www.freecodecamp.org/news/node-version-manager-nvm-install-guide/)](url)

**Comment by xtx29** [comment]
Solved this by installing the latest version of node. Here's how to upgrade node version as mentioned [here](https://askubuntu.com/questions/1152570/npm-cant-find-module-semver-error-in-ubuntu-19-04). It worked on my Ubuntu 22.04

To check your Ubuntu version: `cat /etc/lsb-release`

```
sudo rm -rf /usr/local/bin/npm /usr/local/share/man/man1/node* ~/.npm
sudo rm -rf /usr/local/lib/node*
sudo rm -rf /usr/local/bin/node*
sudo rm -rf /usr/local/include/node*

sudo apt-get purge nodejs npm
sudo apt autoremove
```

Download the latest tar.xz NodeJS file from https://nodejs.org/en/download/

```
tar -xf node-v#.#.#-linux-x64.tar.xz
sudo mv node-v#.#.#-linux-x64/bin/* /usr/local/bin/
sudo mv node-v#.#.#-linux-x64/lib/node_modules/ /usr/local/lib/
```

_Where #.#.# is the version you downloaded._

Verify installation using

```
node -v
npm -v
```

**Reply by ghost** [reply]
does this works in mint?

**Comment by TulossSolutions** [comment]
Hello,

I was able to fix the error by upgrading NodeJS to the latest stable version.

`sudo npm install -g n`

`sudo n stable`

then
    `npx tailwindcss -i ./path/style.css -o ./path/output.css --watch`

**Comment by yigitortu** [comment]
Hello
i just download new npm version and it's worked 
here is my npm versions
![image](https://github.com/tailwindlabs/tailwindcss/assets/72730124/216b1741-a8d7-4d8c-8f5a-e27b94db6187)

**Comment by johnhkchen** [comment]
For those who are googling this question, it might be helpful to make sure your `tailwind.config.js` file is correctly formatted.

In my case, my issue was not wrapping my `content` config in a string. `content: [../templates/**/*.{html,js}],` -> `content: ["../templates/**/*.{html,js"}],`

**Comment by Icegreeen** [comment]
Error: `**SyntaxError: invalid or unexpected token**`

In my case, I found the solution. It was for the following issue:

- **`I was importing Tailwind incorrectly into one of my components.`**

See the image below. I simply **`removed this import`** and the build error no longer occurred.

![Captura de tela de 2024-07-12 20-29-05](https://github.com/user-attachments/assets/f6074a66-7057-4d50-8d68-8c3c1c937114)

Solution: `Check your project to see if there is any improper import in your components or pages in the project.`

Version
```
"next": "^14.2.3",
"tailwindcss": "3.4.1",
```

**Comment by ooyay** [comment]
in my case there was syntax error in tailwind config file.
